### ISSUE TITLE
guided napari UI "tour" 

### ISSUE THREAD

**Issue by tlambert03** [napari/napari#2361]
## 🚀 Feature
Now that we have settings, and we know whether it's the user's first launch. We could conceivably add a quick "guided tour" of the UI. (you know, where arrows point at things and walk you through a couple basics, adding layers, etc...)

I was reminded of this by using Qt Creator which has a guided tour... their source code for that is [here](https://github.com/qt-creator/qt-creator/blob/48c56416f5e128c04f2d2d867b518775e0527607/src/plugins/welcome/introductionwidget.cpp#L72), which might provide some tricks.

Assuming people thing this is a good idea... what should the tour do?? :)
I was thinking:

- highlight a few general areas of the viewer, including the layer list
- prompt them to add a layer probably an image first, that we would add when they click "add an image" or something
- introduce them to the Controls, changing clims and colormap for instance
- have them add a points layer
- show them the points "mode" buttons and add a few new points
- toggle visibility
- show 3d?

thoughts?

**Comment by tlambert03** [comment]
(here's kind of what I mean) ... cc @liaprins-czi who can give us something nicer to shoot for :)

![Untitled](https://user-images.githubusercontent.com/1609449/110216607-092f3580-7e7e-11eb-8823-aa04ae8dbcfa.png)

**Comment by sofroniewn** [comment]
I love this idea, curious if @LCObus has ideas too! It would also be nice to resurrect our welcome visual (we had to hide it here #1922 because of bugs, and should probably redo in Qt and not vispy). Maybe @ppwadhwa can take that on soon :-)

**Comment by LCObus** [comment]
I also love this idea! There are lots of sites/platforms that do this really well- I like when they are:
1. light (aka you can dismiss easily if you dont want to go through it all, or be shown a function but not necessarily have to perform it) 
2. findable (it collapses into a top right icon that can be revisited if needed, or expands into a broader help section) 
3. coordinated with documentation (ie each arrow nudge should map in order to how the welcome tutorial is written, so a user could follow along)

**Comment by goanpeca** [comment]
This is a great idea, I actually implemented the tours in Spyder which included, fading and highlighting of sections, plus the ability to run code.

**Comment by tlambert03** [comment]
> I actually implemented the tours in Spyder which included, fading and highlighting of sections, plus the ability to run code.

fantastic!  I nominate you! 😃

**Comment by ppwadhwa** [comment]
@sofroniewn do you want the welcome screen update in addition to the UI tour?  Or will the UI tour end up including that?

**Comment by sofroniewn** [comment]
> @sofroniewn do you want the welcome screen update in addition to the UI tour? Or will the UI tour end up including that?

I think the welcome screen is probably separate, in the way that tools like VSCode always show something when empty - was just something I thought of again in this context. See #1132 for the original issue. We had some struggles with contrast levels and bugs though, which is why they the screen is turned off. We could make a new issue for getting it working again.

**Comment by sdiebolt** [comment]
With #8995 now merged, this issue might be good to close?

**Comment by TimMonko** [comment]
I agree! Especially because I think your implementation hit all the discussion points in this thread, including the way it highlights, closes, and has a welcome screen tip. Thanks for bumping this!
